rest_api/test/test_rest_api.py

```diff
@@ -265,22 +233,24 @@ def test_file_upload(client: TestClient, api_document_store: BaseDocumentStore):
     response = client.post(url="/file-upload", files=file_to_upload, data={"meta": '{"test_key": "test_value"}'})
     assert 200 == response.status_code
 
-    documents = api_document_store.get_all_documents()
-    assert len(documents) > 0
-    for doc in documents:
-        assert doc.meta["name"] == "sample_pdf_1.pdf"
-        assert doc.meta["test_key"] == "test_value"
+    # ensure the converter was called with the right params
+    _, kwargs = MockPDFToTextConverter.mocker.call_args
+    # Files will be converted to something like 83f4c1f5b2bd43f2af35923b9408076b_sample_pdf_1.pdf
+    # so we ensure the original file name is contained in the converted file name
+    assert "sample_pdf_1.pdf" in kwargs["file_path"].parts[-1]
+    assert kwargs["meta"]["test_key"] == "test_value"
+    MockPDFToTextConverter.mocker.reset_mock()
```

Comment:
Personal opinion: the new test is much harder to read. I hope by using the `TextConverter` we can go back to the simpler version.

Reply:
You're right, this is a better way to test the API only. I don't know how to improve this, but for example the `reset_mock()` call looks like something that can go in a teardown method instead of the body of the test... I'll let you decide if and how this test can be improved though.